source/base/timer.cc

```diff
@@ -1018,6 +1018,7 @@ void
 TimerOutput::disable_output()
 {
   output_is_enabled = false;
+  out_stream.set_condition(false);
```

Comment:
I don't think this is right. What `TimerOutput::enable/disable_output()` does is determine whether the `TimerOutput` object wants to output something. Whether the underlying `out_stream` object actually wants to show the resulting text is a separate question, unrelated to what the current function does. In particular, if you call `TimerOutput::disable_output()`, then with your patch you would also suppress output *from all other places* that might write into the `ConditionalOStream` object. That is not right.

Reply:
Well, first of all, the `out_stream` is owned by the `TimerOutput` itself and so enable/disable it **would not** effect the outer `ConditionalOStream`. It only copies the input condition and construct its own, or set to active if input a `ostream`. And to your first concern, if the timer itself doesn't want to output anything, it won't harm if you set its own `out_stream` to inactive.

Reply:
If you look into the source file of [timer.cc](https://github.com/dealii/dealii/blob/master/source/base/timer.cc), `output_is_enabled` is only checked when `~TimerOutput()` is invoked, and thus `disable/enable_output` only works when you construct it and let it print when destruct it.

But in most time, I need to output it multiple times by manually calling `print_wall_time_statistics()` or `print_summary()`. They do not check this flag and it is hard to add so (also contains MPI func, cannot simply add a high-level check).

So the better way is just deactivating the `out_stream` since we don't need the output.

Reply:
I see. I agree that the design of this class is pretty broken in this regard. Thanks for debugging it and for patiently explaining it to me!